File: avue-helper/src/main/java/com/example/avuehelper/exchange/api/DsWorkerExchange.java
package com.example.avuehelper.exchange.api;

import com.example.avuehelper.exchange.request.ExecSelectSQLRequest;
import com.example.avuehelper.exchange.response.DsWorkerRespJSON;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.service.annotation.HttpExchange;
import org.springframework.web.service.annotation.PostExchange;

import java.util.List;
import java.util.Map;

@HttpExchange("/ds-worker")
public interface DsWorkerExchange {

    @PostExchange("/exec/select")
    DsWorkerRespJSON<List<Map<String, Object>>> execSelectSQL(@RequestBody ExecSelectSQLRequest body);
}

File: chat-analysis/src/main/kotlin/org/inet/aet/chatanalysis/exchange/api/DsWorkerExchange.kt
package org.inet.aet.chatanalysis.exchange.api

import org.inet.aet.chatanalysis.entity.DBSchema
import org.inet.aet.chatanalysis.exchange.request.DBSchemaRequest
import org.inet.aet.chatanalysis.exchange.request.ExecSelectSQLRequest
import org.inet.aet.chatanalysis.exchange.response.DsWorkerRespJSON
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.service.annotation.HttpExchange
import org.springframework.web.service.annotation.PostExchange

@HttpExchange("/ds-worker")
interface DsWorkerExchange {

    @PostExchange("/exec/select")
    fun executeSelectSQL(@RequestBody body: ExecSelectSQLRequest): DsWorkerRespJSON<List<Map<String, String>>?>

    @PostExchange("/meta/db-schema")
    fun getDBSchema(@RequestBody body: DBSchemaRequest): DsWorkerRespJSON<DBSchema>
}